enterprise/internal/batches/store/changeset_specs.go

```diff
@@ -107,6 +109,13 @@ func (s *Store) CreateChangesetSpec(ctx context.Context, cs ...*btypes.Changeset
 				}
 			}
 
+			// We check if the resulting diff is greater than 1GB, since the limit
+			// for the diff column (which is bytea) is 1GB
+			if len(c.Diff) > OneGigabyte {
+				link := "https://docs.sourcegraph.com/batch_changes/references/batch_spec_yaml_reference#transformchanges"
+				return errors.Errorf("The changeset generated is over the size limit. You can make use of [transformChanges](%s) to break down the changesets into smaller pieces.", link)
```

Comment:
```suggestion
				return errors.Errorf("The changeset patch generated is over the size limit. You can make use of [transformChanges](%s) to break down the changesets into smaller pieces.", link)
```
Maybe? Might make it a little clearer that it's explicitly the diff being troublesome here.